sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -78,50 +85,55 @@ public NettyAsyncHttpClientBuilder(HttpClient nettyHttpClient) {
-                        case HTTP:
-                            nettyProxy = ProxyProvider.Proxy.HTTP;
-                            break;
-                        case SOCKS4:
-                            nettyProxy = ProxyProvider.Proxy.SOCKS4;
-                            break;
-                        case SOCKS5:
-                            nettyProxy = ProxyProvider.Proxy.SOCKS5;
-                            break;
-                        default:
-                            throw logger.logExceptionAsError(new IllegalStateException(
-                                String.format("Unknown Proxy type '%s' in use. Not configuring Netty proxy.",
-                                    proxyOptions.getType())));
-                    }
-                    if (proxyOptions.getUsername() != null) {
-                        // Netty supports only Basic proxy authentication and we default to it.
-                        return tcpConfig.proxy(ts -> ts.type(nettyProxy)
-                                .address(proxyOptions.getAddress())
-                                .username(proxyOptions.getUsername())
-                                .password(userName -> proxyOptions.getPassword())
-                                .build());
-                    } else {
-                        return tcpConfig.proxy(ts -> ts.type(nettyProxy).address(proxyOptions.getAddress()));
-                    }
+                ProxyOptions buildProxyOptions = (proxyOptions == null)
+                    ? ProxyOptions.loadFromConfiguration(buildConfiguration)
+                    : proxyOptions;
+
+                if (buildProxyOptions != null) {
+                    tcpConfig = tcpConfig.proxy(typeSpec ->
+                        typeSpec.type(mapProxyType(buildProxyOptions.getType(), logger))
+                            .address(proxyOptions.getAddress())
+                            .username(proxyOptions.getUsername())
+                            .password(user -> proxyOptions.getPassword())
+                            .nonProxyHosts(proxyOptions.getNonProxyHosts()));
                 }
+
                 return tcpConfig;
             });
+
         return new NettyAsyncHttpClient(nettyHttpClient);
     }
 
+    /*
+     * Maps a 'ProxyOptions.Type' to a 'ProxyProvider.Proxy', if the type is unknown or cannot be mapped an
+     * IllegalStateException will be thrown.
+     */
+    private static ProxyProvider.Proxy mapProxyType(ProxyOptions.Type type, ClientLogger logger) {
+        switch (type) {
+            case HTTP:
+                return ProxyProvider.Proxy.HTTP;
+            case SOCKS4:
+                return ProxyProvider.Proxy.SOCKS4;
+            case SOCKS5:
+                return ProxyProvider.Proxy.SOCKS5;
+            default:
+                throw logger.logExceptionAsError(new IllegalStateException(
+                    String.format("Unknown Proxy type '%s' in use. Not configuring Netty proxy.", type)));
```

Comment:
Think of this error message from the users perspective - this will probably require them to do a search for this string and hope they have found a valid result. What additional guidance can you provide in here to help the user solve this without a web search being required?

Reply:
Updated error message